plugins/dialogui/plugin.js

```diff
@@ -151,7 +151,8 @@ CKEDITOR.plugins.add( 'dialogui', {
 								'<label class="cke_dialog_ui_labeled_label' + requiredClass + '" ', ' id="' + _.labelId + '"',
 									( _.inputId ? ' for="' + _.inputId + '"' : '' ),
 									( elementDefinition.labelStyle ? ' style="' + elementDefinition.labelStyle + '"' : '' ) + '>',
-									elementDefinition.label,
+									// If label is required add red asterisk. (#3433)
+									( elementDefinition.required ? elementDefinition.label + '<span style="color:red;">*</span>' : elementDefinition.label ),
```

Comment:
Instead of inlining styling, we could make this feature a bit more extensible and add CSS class instead, like `cke_dialog_ui_labeled_required`. Such class should be defined in our `./skins` (you can check out how `cke_dialog_ui_labeled_label` was defined), so the option is customizable.